### PR TITLE
Fixed cluttered buttons in Theming Control Panel (Issue #212)

### PR DESCRIPTION
**Pull request by rohnsha0**
as said in the issue #212

### CODE REVIEW

**Comment by rohnsha0** [comment]
@jenkins-plone-org please run jobs

**Comment by stevepiercy** [comment]
This PR does not resolve the issue. Please see this comment: https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687

**Comment by rohnsha0** [comment]
> This PR does not resolve the issue. Please see this comment: [plone/Products.CMFPlone#3595 (comment)](https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687)

@stevepiercy changing the div class doesn't make any changing.... that just changes the top margin....

**Comment by rohnsha0** [comment]
image
LEFT: `<div class="btn-group">`
RIGHT: `<div class="m-1">`

**Comment by rohnsha0** [comment]
image
here's the inspector...

**Comment by rohnsha0** [comment]
image
After changes in this PR! @stevepiercy

**Comment by jensens** [comment]
Thanks for touching this! Idea:
At the moment all are blue (primary). But by convention Upload should be green (it creates something). Test and help could be secondary. Also below buttons could be better. Deactivate could be red, clear cache yellow. All small size and a button group might look better too.

**Comment by rohnsha0** [comment]
@jensens okay... I'll try to perform changes as you suggested... 
But I had a question... Our PLONE control panel to date doesn't have many colors...
Implementing each button with some colorful approach... would that look good?

**Comment by jensens** [comment]
> @jensens okay... I'll try to perform changes as you suggested... But I had a question... Our PLONE control panel to date doesn't have many colors... Implementing each button with some colorful approach... would that look good?

We have green at least for creation, should be visible e.g. in users and groups.

**Comment by rohnsha0** [comment]
> 

okay...
Pushing soon...

**Comment by rohnsha0** [comment]
image

@jensens now?

**Comment by stevepiercy** [comment]
> > This PR does not resolve the issue. Please see this comment: [plone/Products.CMFPlone#3595 (comment)](https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687)
> 
> @stevepiercy changing the div class doesn't make any changing.... that just changes the top margin....

The original issue (https://github.com/plone/Products.CMFPlone/issues/3595) says:

> The top 3 buttons have no paddings, the placement of the bottom buttons also needs some love.

You are focusing on the bottom buttons, and ignoring the top buttons. If you apply the change I suggested to the top buttons, it should be fine.

Your target should be to replicate the existing look and feel of the Plone 5.2 UI, which is called Classic UI in Plone 6.

## Screen shot of Theming control panel on Plone 5.2 demo.

https://demo.plone.org/@@theming-controlpanel

Screen Shot 2022-08-23 at 2 15 20 PM

@jensens where is the style guide for the Plone Classic UI? All those color changes to the buttons seems inconsistent to the entire Classic UI. IMO it also looks bad.

**Comment by rohnsha0** [comment]
I got your point @stevepiercy 
Implementing the suggested changes in a while!

**Comment by rohnsha0** [comment]
I too thought that v6b1's theming panel should have the same functionality as v5.2.9
But nevertheless I'll add the required buttons to v6 control panels

**Comment by stevepiercy** [comment]
Please do not _add_ buttons, and limit changes to _spacing_ the _existing_ buttons, as the original issue notes.

If you would like to add functionality with new buttons, then that should be discussed either:

1. In a new and separate issue together with the Classic UI team.
2. In the [Community Forum > Developer > User Interface category](https://community.plone.org/c/development/user-interface/14).
3. In [Discord, #classic-ui channel](https://discord.com/channels/786421998426521600/787254601656827924).

This will save you a lot of potentially wasted effort.

**Comment by rohnsha0** [comment]
image
@stevepiercy I've added some buttons... and changed the buttons appearance too...

**Comment by rohnsha0** [comment]
but if you say I'll limit the changes...
and just push the changed `appearance` and your suggested `div class`

**Review by mauritsvanrees — COMMENTED** [review]
The buttons that are missing, have been removed explicitly because they are not supported anymore in Plone. This was part of a big refactoring and simplification of the resource registry.

With your fix this already looks better. Thanks!

I think it would look even better if the four buttons were evenly on two rows. And then center the buttons, so they line out with each other. I am thinking of something like this:

       Update   | Deactivate
    Clear Cache |  Download

Is that something you can try?

**Comment by rohnsha0** [comment]
@mauritsvanrees In my opinion, `UPDATE`, `DEACTIVATE`, `CLEAR CACHE`, `DOWNLOAD` in a single row would look much nicer...

**Comment by mauritsvanrees** [comment]
If that fits, that would be fine too.

**Comment by rohnsha0** [comment]
> If that fits, that would be fine too.

image
I hope this is good!

**Comment by rohnsha0** [comment]
image
Full screen

**Comment by rohnsha0** [comment]
image

**Comment by stevepiercy** [comment]
You did not implement the change as I directed in https://github.com/plone/Products.CMFPlone/issues/3595#issuecomment-1219026687. See [line 100](https://github.com/plone/plone.app.theming/blob/master/src/plone/app/theming/browser/controlpanel.pt#L100)

**Comment by rohnsha0** [comment]
> You did not implement the change as I directed in [plone/Products.CMFPlone#3595 (comment)

Actually @stevepiercy , I haven't pushed anything since our last convo...
But were discussing about changes in the buttons

File: src/plone/app/theming/browser/controlpanel.pt [hunk 1]
```diff
@@ -181,7 +181,7 @@
 
                             <form method="get" target="_blank" tal:attributes="action string:${context/absolute_url}/++theme++${theme/name}/@@download-zip">
                                 <button i18n:translate=""
-                                    class="btn btn-primary">Download</button>
+                                    class="btn btn-primary mt-1 px-3">Download</button>
```
Comment (stevepiercy) [inline src/plone/app/theming/browser/controlpanel.pt:184]: A class should not be applied to a single button that resides in a div that contains a group of buttons. Instead the [parent div of the group of buttons](https://github.com/plone/plone.app.theming/pull/213/files#diff-c96bcee60e1865b90e3c96e35ed68375a88a71130e599b8ec757dec76a5f0634R151) should have a class applied, specifically `d-grid gap-1`. See https://getbootstrap.com/docs/5.2/utilities/spacing/#gap.

```suggestion
                                    class="btn btn-primary">Download</button>
```

File: src/plone/app/theming/browser/controlpanel.pt [hunk 2]
```diff
@@ -97,7 +97,7 @@
             or select an existing theme from a the list below.
         </p>
 
-        <div class="btn-group">
+        <div class="m-1">
```
Comment (stevepiercy) [inline src/plone/app/theming/browser/controlpanel.pt:100]: Thank you. This is correct.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: src/plone/app/theming/browser/controlpanel.pt (hunk 1 above)
Comment (rohnsha0) [reply src/plone/app/theming/browser/controlpanel.pt:184]: actually, @stevepiercy I've done some slight changes to the block placement and don't need grid spacings
image

Comment (stevepiercy) [reply src/plone/app/theming/browser/controlpanel.pt:184]: Those changes do not appear in this PR. Please commit and push the changes.

Comment (rohnsha0) [reply src/plone/app/theming/browser/controlpanel.pt]: @stevepiercy @mauritsvanrees  Committed changes... kindly have a look!

**Comment by rohnsha0** [comment]
@jenkins-plone-org please run jobs

File: news/212.bugfix [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Fixed/ Revamped cluttered action buttons in Theming Control Panel
```
Comment (ale-rt) [inline news/212.bugfix:1]:
```suggestion
The action buttons in the theming control panel have been improved
```

**Review by ale-rt — APPROVED** [review]
For me this is a nice improvement:
image
I am +1 merging this.

If later we want to improve/refine there is always time.

Thanks a lot for your contribution :)

**Review by mauritsvanrees — APPROVED** [review]

File: news/212.bugfix (hunk 3 above)
Comment (rohnsha0) [reply news/212.bugfix:1]: @ale-rt sorry' didn't noticed this one ..
I'll edit this one directly! 😑

**Pull request merged**